src/main/java/dev/dkvl/womutils/WomUtilsPlugin.java

```diff
@@ -205,6 +249,259 @@ public void onResponse(Call call, Response response)
+			{
+				return;
+			}
+
+			final MenuEntry addMember = new MenuEntry();
+			if (groupMembers.contains(Text.toJagexName(Text.removeTags(target).toLowerCase())))
+			{
+				addMember.setOption(REMOVE_MEMBER);
+			} else {
+				addMember.setOption(ADD_MEMBER);
+			}
+			addMember.setTarget(target);
+			addMember.setType(MenuAction.RUNELITE.getId());
+			addMember.setParam0(event.getActionParam0());
+			addMember.setParam1(event.getActionParam1());
+			addMember.setIdentifier(event.getIdentifier());
+
+			insertMenuEntry(addMember, client.getMenuEntries());
+		}
+	}
+
+	@Subscribe
+	public void onPlayerMenuOptionClicked(PlayerMenuOptionClicked event)
+	{
+		String target = Text.toJagexName(event.getMenuTarget()).toLowerCase();
+		if (event.getMenuOption().equals(ADD_MEMBER))
+		{
+			addGroupMember(target);
+		}
+		else if (event.getMenuOption().equals(REMOVE_MEMBER))
+		{
+			removeGroupMember(target);
+		}
+	}
+
+	@Subscribe
+	public void onWidgetMenuOptionClicked(final WidgetMenuOptionClicked event)
+	{
+		WidgetInfo widget = event.getWidget();
+		if (widget == WidgetInfo.FIXED_VIEWPORT_FRIENDS_CHAT_TAB
+				|| widget == WidgetInfo.RESIZABLE_VIEWPORT_FRIENDS_CHAT_TAB)
+		{
+			importGroupMembers();
+		}
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("womutils"))
+		{
+			if (config.menuOptions()) {
+				addImportMenuOption();
+			} else {
+				removeImportMenuOption();
+			}
+		}
+	}
+
+
+	private void insertMenuEntry(MenuEntry newEntry, MenuEntry[] entries)
+	{
+		MenuEntry[] newMenu = ObjectArrays.concat(entries, newEntry);
+		int menuEntryCount = newMenu.length;
+		ArrayUtils.swap(newMenu,  menuEntryCount - 1, menuEntryCount - 2);
+		client.setMenuEntries(newMenu);
+	}
+
+	private void addImportMenuOption()
+	{
+		menuManager.addManagedCustomMenu(FIXED_FRIENDS_TAB_IMPORT);
+		menuManager.addManagedCustomMenu(RESIZABLE_FRIENDS_TAB_IMPORT);
+	}
+
+	private void removeImportMenuOption()
+	{
+		menuManager.removeManagedCustomMenu(FIXED_FRIENDS_TAB_IMPORT);
+		menuManager.removeManagedCustomMenu(RESIZABLE_FRIENDS_TAB_IMPORT);
+	}
+
+	private void addGroupMember(String username)
+	{
+		ArrayList<JsonObject> members = new ArrayList<>();
+		JsonObject member = new JsonObject();
+		member.addProperty("username", username);
+		member.addProperty("role", "member");
+		members.add(member);
+
+		GroupMemberAddition gme = new GroupMemberAddition(config.verificationCode(), members);
+
+		HttpUrl url = new HttpUrl.Builder()
+				.scheme("https")
+				.host("api.wiseoldman.net")
+				.addPathSegment("groups")
+				.addPathSegment(String.valueOf(config.groupId()))
+				.addPathSegment("add-members")
+				.build();
+
+		String payload = gson.toJson(gme);
+
+		RequestBody body = RequestBody.create(
+				MediaType.parse("application/json; charset=utf-8"),
+				payload
+		);
+
+		Request request = new Request.Builder()
+				.header("User-Agent", "RuneLite")
+				.url(url)
+				.post(body)
+				.build();
```

Comment:
We should consider moving the request-building into a single method, right now it's 20 lines that have to be copy pasted with a diff of 1-2 lines.